go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -475,18 +475,20 @@ func (tm *TabletManager) demotePrimary(ctx context.Context, revertPartialFailure
 		}
 	}()
 
-	// If using semi-sync, we need to disable primary-side.
-	if err := tm.fixSemiSync(topodatapb.TabletType_REPLICA); err != nil {
-		return nil, err
-	}
-	defer func() {
-		if finalErr != nil && revertPartialFailure && wasPrimary {
-			// enable primary-side semi-sync again
-			if err := tm.fixSemiSync(topodatapb.TabletType_PRIMARY); err != nil {
-				log.Warningf("fixSemiSync(PRIMARY) failed during revert: %v", err)
-			}
+	if tm.isPrimarySideSemiSyncEnabled() {
```

Comment:
Can you add a comment here as to why we should check this first?

Reply:
Added